src/adhocracy_core/adhocracy_core/content/__init__.py

```diff
@@ -326,6 +327,14 @@ def can_delete_anonymized(self, context: object, request: Request) -> bool:
         can_anonymize = _is_anonymized_and_has_permission(context, request)
         return can_anonymize
 
+    def is_password_required(self, context: object, request: Request) -> bool:
+        """Check if some sheets of `context` require a password for editing."""
+        sheets_edit = self.get_sheets_edit(context, request)
+        sheets_require_password = \
+            [s for s in sheets_edit
+             if s.meta.isheet.isOrExtends(ISheetRequirePassword)]
+        return bool(sheets_require_password)
```

Comment:
`any((s.meta.isheet.isOrExtends(ISheetRequirePassword) for s in sheets_edit))` would be more efficient.